### PR TITLE
Vof dcdd improvement

### PR DESCRIPTION
**Pull request by blaisb**
# Description of the problem

- The previous Shock capturing scheme was not formulated correctly

# Description of the solution

- I tried different formulation, but it seems the DCDD shock capturing scheme is a very good compromise between too much additional viscosity (what other shock capture schemes do) and robustness
- Implemented the fixed formulation which is the correct one
- Added the references to the correct body of literature.

# How Has This Been Tested?

- Application tests have slightly changed
- Reran a lot of the examples and they give very good results.

# Documentation

- During the Hackathon will work on making a full documentation for it.

### CODE REVIEW

**Review by AmishgaAlphonius — APPROVED** [review]
I only have the same comment as before (to explicit what DCDD stands for) :)

File: source/solvers/vof_assemblers.cc [hunk 1]
```diff
@@ -52,23 +52,25 @@ VOFAssemblerCore<dim>::assemble_matrix(VOFScratchData<dim> &      scratch_data,
...
-      // Gather the order of the VOF interpolation
-      const double order = this->fem_parameters.VOF_order;
-
-      // Calculate the artificial viscosity of the shock capture
-      const double vdcdd = (0.5 * h) * (velocity.norm() * velocity.norm()) *
-                           pow(phase_gradient_norm * h, order);
+      // Tezduyar, T. E. (2003). Computation of moving boundaries and interfaces
+      // and stabilization parameters. International Journal for Numerical
+      // Methods in Fluids, 43(5), 555-575. We implement the equivalent of
+      // equation (70) and (79)
```
Comment (hepap) [inline source/solvers/vof_assemblers.cc:58]:
```suggestion
      // Methods in Fluids, 43(5), 555-575. Our implementation is based on 
      // equations (70) and (79), which are adapted for the VOF solver.
```

File: source/solvers/vof_assemblers.cc [hunk 2]
```diff
@@ -52,23 +52,25 @@ VOFAssemblerCore<dim>::assemble_matrix(VOFScratchData<dim> &      scratch_data,
...
 
       const double tolerance = 1e-12;
 
+      // In Tezduyar 2003 this is noted r
```
Comment (hepap) [inline source/solvers/vof_assemblers.cc:62]:
```suggestion
      // In Tezduyar 2003, this is denoted r
```

File: source/solvers/vof_assemblers.cc [hunk 3]
```diff
@@ -52,23 +52,25 @@ VOFAssemblerCore<dim>::assemble_matrix(VOFScratchData<dim> &      scratch_data,
...
       Tensor<1, dim> gradient_unit_vector =
         phase_gradient / (phase_gradient_norm + tolerance);
 
+      // Calculate the artificial viscosity of the shock capture
+      const double vdcdd =
+        (0.5 * h * h) * velocity.norm() * phase_gradient_norm;
+      //(0.5 * h) * (velocity.norm() * velocity.norm()) *
+      // pow(phase_gradient_norm * h, order);
+
+
```
Comment (hepap) [inline source/solvers/vof_assemblers.cc:72]:
```suggestion
```

File: source/solvers/vof_assemblers.cc [hunk 4]
```diff
@@ -198,23 +200,21 @@ VOFAssemblerCore<dim>::assemble_rhs(VOFScratchData<dim> &      scratch_data,
...
-      // Gather the order of the VOF interpolation
-      const double order = this->fem_parameters.VOF_order;
-
-      // Calculate the artificial viscosity of the shock capture
-      const double vdcdd = (0.5 * h) * (velocity.norm() * velocity.norm()) *
-                           pow(phase_gradient.norm() * h, order);
+      // Tezduyar, T. E. (2003). Computation of moving boundaries and interfaces
+      // and stabilization parameters. International Journal for Numerical
+      // Methods in Fluids, 43(5), 555-575. We implement the equivalent of
+      // equation (70) and (79)
```
Comment (hepap) [inline source/solvers/vof_assemblers.cc:206]:
```suggestion
      // Methods in Fluids, 43(5), 555-575. Our implementation is based on 
      // equations (70) and (79), which are adapted for the VOF solver.
```

File: source/solvers/vof_assemblers.cc [hunk 5]
```diff
@@ -198,23 +200,21 @@ VOFAssemblerCore<dim>::assemble_rhs(VOFScratchData<dim> &      scratch_data,
...
 
       const double tolerance = 1e-12;
 
+      // In Tezduyar 2003 this is noted r
```
Comment (hepap) [inline source/solvers/vof_assemblers.cc:210]:
```suggestion
      // In Tezduyar 2003, this is denoted r
```

File: source/solvers/vof_assemblers.cc [hunk 6]
```diff
@@ -198,23 +200,21 @@ VOFAssemblerCore<dim>::assemble_rhs(VOFScratchData<dim> &      scratch_data,
...
       Tensor<1, dim> gradient_unit_vector =
         phase_gradient / (phase_gradient_norm + tolerance);
 
+      // Calculate the artificial viscosity of the shock capture
+      const double vdcdd =
+        (0.5 * h * h) * velocity.norm() * phase_gradient_norm;
+
       // We neglect to remove the diffusion aligned with the velocity
       // as is done in the original article. This term generates poorer
       // results than just using the vdcdd approach.
```
Comment (hepap) [inline source/solvers/vof_assemblers.cc:220]: Is this still true?

**Review by hepap — APPROVED** [review]

File: source/solvers/vof_assemblers.cc (hunk 6 above)
Comment (blaisb) [reply source/solvers/vof_assemblers.cc:220]: Yup, will be the next step in a next PR

Comment (hepap) [reply source/solvers/vof_assemblers.cc:220]: In that case,  is this more accurate? 
```suggestion
      // We neglect to remove the diffusion aligned with the velocity
      // as is done in the original article.
```

**Pull request merged**